p11-kit/modules.c

```diff
@@ -185,6 +186,51 @@ const char *p11_config_user_modules = P11_USER_CONFIG_MODULES;
  * P11-KIT FUNCTIONALITY
  */
 
+/**
+ * p11_kit_override_system_files:
+ * @system_conf: the system configuration file (default: system_config_dir/pkcs11.conf)
+ * @user_conf: the user configuration file (default: ~/.config/pkcs11/pkcs11.conf)
+ * @package_modules: location of modules shipped by p11-kit (default: system_config/modules)
+ * @system_modules: location of system pkcs11 modules (default: system_config/modules)
+ * @user_modules: location of user modules (default: ~/.config/pkcs11/modules)
+ *
+ * Overrides the default system configuration files. The
+ * provided values should be accessible for the lifetime
+ * of p11-kit usage.
+ *
+ * When the value %NULL is provided for any of the locations,
+ * it will not be updated.
+ *
+ * This is function intended to be used in test suites and
+ * not production, and as such %P11_KIT_FUTURE_UNSTABLE_API
+ * must be defined before including p11-kit.h.
+ *
+ * Since: 0.23.10
+ *
+ */
+void
+p11_kit_override_system_files (const char *system_conf,
+                               const char *user_conf,
+                               const char *package_modules,
+                               const char *system_modules,
+                               const char *user_modules)
+{
+	assert (system_conf);
```

Comment:
This is causing a test failure in travis:
https://travis-ci.org/p11-glue/p11-kit/jobs/295657233#L4820
I guess this (and the following assignments) should be something like:
```
if (system_conf)
        p11_config_system_file = system_conf;
```

Reply:
fixed. sorry for that.